### ISSUE TITLE
Block address shortcut

### ISSUE THREAD

**Issue by xarg** [leggett/simplify#652]
Currently the "mark as spam" shortcut works as expected with `!`, however Gmail does not have a native "block address" shortcut as shown in the screenshot below. It would be amazing to have this shortcut to permanently block an email address.

As a bonus (probably beyond the scope of Simplify), I know that Superhuman has a block domain shortcut which creates a filter on the entire domain rather than a specific address which is useful too.

![image](https://user-images.githubusercontent.com/94721/149177750-32061f05-47b0-4cae-a00c-e7a1f9df20ed.png)

I'm willing to pay extra to get the block keyboard shortcut faster, I'm already on a payed plan. Let me know if it's possible. Thank you!

---
Simplify v.2.5.8 - Configuration: aAX appMenuOpen boldHighlight ca3 caI caL cfC cfO cfQ cfR cfU composeActions darkTheme dateGroup fontList fontMsg hideListCount hideMsgCount hideTabIcons htmlEmail inInbox inList invertAddons invertCompose izBg2 js-focus-visible listWidthLg lowDensity minimizeChat msgOpen msgWidthSm navOpen newUI simplify simplifyMenuOpen supExtWarn textButtons vPane - System: Mozilla/5.0 (Macintosh Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/97.0.4692.71 Safari/537.36 - Window: 3440 x 1415 (110pct zoom) - Language: en

**Comment by leggett** [comment]
Makes sense... I've got plans for an action menu like Superhuman where you can type in actions. Can add long-tail things like "block domain" which steps through creating the filter.

For a keyboard shortcut to block this sender... what's a good shortcut? Maybe `@` since it is between `!` (report spam) and `#` (delete)? Both of those are kind of jank shortcuts though. 

Should it also report as spam?

**Comment by xarg** [comment]
> Makes sense... I've got plans for an action menu like Superhuman where you can type in actions. Can add long-tail things like "block domain" which steps through creating the filter.

Great news!
 
> For a keyboard shortcut to block this sender... what's a good shortcut? Maybe `@` since it is between `!` (report spam) and `#` (delete)? 

`@` good choice if it's not used. There will be a confirmation anyway.

> Both of those are kind of jank shortcuts though.

IMO, it's better than nothing.

> Should it also report as spam?

Yes, I'd say it should + unsubscribe (ideally) so the domain doesn't get even worse reputation if they actually took some time to implement the unsubscribe function.

**Comment by leggett** [comment]
Ok, I tried to add this and forgot about a major blocker to adding these kinds of keyboard shortcuts.

**Brief background:**

- Most of Simplify's keyboard shortcuts work by simulating a click on the corresponding button in the interface
- The browser only lets you do this as an extension when the user interacted with the app in some way (e.g. pressed a key)
- One interaction = one click. You can't click on different elements from one user interaction
- Actions hidden in menus require two clicks. So when I went to build this, I could only get the message menu to open and it wouldn't click on the "Block sender" action
- Gmail doesn't add the menu to the DOM until you click on the menu button, so I have to click on the menu first

**Options**
So, to do this, it would need to be a sequential key shortcut like `Shift+.` (open the message menu) → `b` (block)

The only other option I could think of (but wasn't able to get to work) is Gmail supports mouse down → move mouse to menu item → release mouse to open a menu and select something in it. But I wasn't able to simulate that with MouseEvents.

**Comment by leggett** [comment]
Upon more digging, I don't think that is what is blocking it. Something else isn't working... will keep digging.

**Comment by xarg** [comment]
Thanks @leggett! Didn't expect you to jump right on it 😀

Let me know if I can help. Two shortcuts in sequence is not ideal, but I would still use it.
